posthog/queries/insight.py

```diff
@@ -0,0 +1,30 @@
+from typing import Optional
+
+from posthog.clickhouse.query_tagging import tag_queries
+from posthog.client import sync_execute
+from posthog.types import FilterType
+
+
+# Wrapper around sync_execute, adding query tags for insights performance
+def insight_sync_execute(
```

Comment:
Could nitpick at the name ... It's about tagging the query rather than insights but more important to start gathering the data than to get the names right immediately 

But, lovely separation of this behaviour over having a conditional inside sync_execute

Reply:
Naming is hard and I indeed struggled here. Suggestions?